src/rod/sdf/sdf.py

```diff
@@ -60,34 +60,30 @@ def load(sdf: pathlib.Path | str, is_urdf: bool | None = None) -> Sdf:
             The parsed SDF file.
         """
 
-        # Handle the max path length depending on the OS
+        # Handle the max path length depending on the OS.
         try:
             from ctypes.wintypes import MAX_PATH
         except ValueError:
             MAX_PATH = os.pathconf("/", "PC_PATH_MAX")
 
-        if is_urdf is not None:
-            sdf_string = sdf
-        else:
-            match sdf:
-                # Case 1: It's a Path object
-                case pathlib.Path():
-                    sdf_string = sdf.read_text()
-                    is_urdf = sdf.suffix == ".urdf"
-
-                # Case 2: It's a string with a path
-                case str() if len(sdf) <= MAX_PATH and pathlib.Path(sdf).is_file():
-                    sdf_string = pathlib.Path(sdf).read_text(encoding="utf-8")
-                    is_urdf = pathlib.Path(sdf).suffix == ".urdf"
-
-                # Case 3: It's an SDF/URDF string
-                case str():
-                    sdf_string = sdf
-                    is_urdf = "<robot" in sdf_string
-
-                # Case 4: Raise an error for unsupported types
-                case _:
-                    raise TypeError(f"Unsupported type for 'sdf': {type(sdf)}")
+        # Check if the input is a string path.
+        is_str_path = getattr(sdf, "__len__", lambda: MAX_PATH + 1)() <= MAX_PATH
+
+        match sdf:
+            # Case 1: Handle SDF/URDF string.
+            case str() if not is_str_path:
+                sdf_string = sdf
+                is_urdf = is_urdf if is_urdf is not None else "<robot" in sdf_string
```

Comment:
There could be SDF/URDF content shorter than MAX_PATH=260 characters right?

Reply:
Solved by rebase